skimage/util/_flood_fill.py

```diff
@@ -0,0 +1,182 @@
+           [5, 0, 0, 0, 0, 0, 3]])
+
+    Fill connected ones with 5, with only cardinal direction connectivity:
+
+    >>> flood_fill(image, (1, 1), 5, connectivity=1)
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [1, 0, 0, 0, 0, 0, 3]])
+
+    Fill with a tolerance:
+
+    >>> flood_fill(image, (0, 0), 5, tolerance=1)
+    array([[5, 5, 5, 5, 5, 5, 5],
+           [5, 5, 5, 5, 2, 2, 5],
+           [5, 5, 5, 5, 2, 2, 5],
+           [5, 5, 5, 5, 5, 5, 3]])
+    """
+    from ..morphology.extrema import (_resolve_neighborhood,
+                                      _set_edge_values_inplace, _fast_pad)
+    from ..morphology.watershed import _offsets_to_raveled_neighbors
+    from ._flood_fill_cy import _flood_fill
+
+    image = np.asarray(image, order='C')
+    seed_value = image[seed_point]
+
+    # Shortcut for rank zero
+    if image.size == 0:
+        return np.array([], dtype=(np.intp if indices else np.uint8))
+
+    # Convenience for 1d input
+    try:
+        iter(seed_point)
+    except TypeError:
+        seed_point = (seed_point,)
+
+    selem = _resolve_neighborhood(selem, connectivity, image.ndim)
+
+    # Must annotate borders
+    working_image = _fast_pad(image, image.min())
+
+    # Correct start point in ravelled image
+    ravelled_seed_idx = np.ravel_multi_index([i+1 for i in seed_point],
+                                             working_image.shape)
+    neighbor_offsets = _offsets_to_raveled_neighbors(
+        working_image.shape, selem, center=((1,) * image.ndim))
+
+    # Use a set of flags; see _flood_fil.pyx for meanings
+    flags = np.zeros(working_image.shape, dtype=np.uint8)
+    _set_edge_values_inplace(flags, value=2)
+
+    # Inform the Cython function which path to take
+    if tolerance is not None:
+        do_tol = 1
+        # Check if tolerance could create overflow problems
+        try:
+            max_value = np.finfo(working_image.dtype).max
+            min_value = np.finfo(working_image.dtype).min
+        except ValueError:
+            max_value = np.iinfo(working_image.dtype).max
+            min_value = np.iinfo(working_image.dtype).min
+
+        high_tol = min(max_value, seed_value + tolerance)
+        low_tol = max(min_value, seed_value - tolerance)
+
+    else:
+        do_tol = 0
+        high_tol = seed_value
+        low_tol = seed_value
+
+    # Run flood fill
+    try:
+        _flood_fill(working_image.ravel(), flags.ravel(), neighbor_offsets,
+                    ravelled_seed_idx, seed_value, do_tol, high_tol, low_tol)
+    except TypeError:
+        if working_image.dtype == np.float16:
+            # Provide the user with clearer error message
+            raise TypeError("dtype of `image` is float16 which is not "
+                            "supported, try upcasting to float32")
+        else:
+            raise
+
+    original_slice = tuple(slice(1, -1) for _ in range(image.ndim))
```

Comment:
Even if it may not matter in the big picture `(slice(1, -1),) * image.ndim` is faster and scales better.